### PR TITLE
Add support for pre-built and released contract deploys

### PR DESCRIPTION
**Pull request by jarrodwatts**

### CODE REVIEW

File: src/cli/index.ts [hunk 1]
```diff
@@ -90,8 +91,37 @@ $$$$$$\\   $$$$$$$\\  $$\\  $$$$$$\\   $$$$$$$ |$$\\  $$\\  $$\\  $$$$$$\\  $$$$
     .option("--dry-run", "dry run (skip actually publishing)")
     .option("-d, --debug", "show debug logs")
     .option("--ci", "Continuous Integration mode")
+    .option(
+      "-n, --name [name]",
+      "Name of the pre-built or released contract (such as nft-drop)",
+    )
+    .option(
+      "--cv, --contract-version [version]",
```
Comment (joaquim-verges) [inline src/cli/index.ts:99]: I'm guessing we can't use --version here?

File: src/cli/index.ts [hunk 2]
```diff
@@ -90,8 +91,37 @@ $$$$$$\\   $$$$$$$\\  $$\\  $$$$$$\\   $$$$$$$ |$$\\  $$\\  $$\\  $$$$$$\\  $$$$
...
+              `Could not find a contract named ${options.name} ${
+                options.contractVersion || ""
+              }`,
+            ),
+          );
+          return;
+        }
+
+        info(`Open this link to deploy your contract:`);
+        logger.info(chalk.blueBright(url));
```
Comment (joaquim-verges) [inline src/cli/index.ts:118]: This should also open the browser automatically like we do for deploy/release

File: src/cli/index.ts [hunk 3]
```diff
@@ -90,8 +91,37 @@ $$$$$$\\   $$$$$$$\\  $$\\  $$$$$$\\   $$$$$$$ |$$\\  $$\\  $$\\  $$$$$$\\  $$$$
...
+    .option(
+      "--cv, --contract-version [version]",
+      "Version of the released contract",
+    )
     .action(async (options) => {
-      const url = await processProject(options, "deploy");
+      let url;
+
+      if (options.name) {
+        url = generateDashboardUrl(options.name, options.contractVersion);
```
Comment (joaquim-verges) [inline src/cli/index.ts]: Could be a const, just use a diff variable name

File: src/helpers/generate-dashboard-url.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+export default function generateDashboardUrl(
+  name: string,
+  version: string | undefined,
+): string | undefined {
+  const splitName = name.split("/");
+  // Released contract
+  if (splitName.length > 1) {
+    return `https://thirdweb.com/dashboard/${splitName[0]}/${splitName[1]}/${
```
Comment (joaquim-verges) [inline src/helpers/generate-dashboard-url.ts]: Nitpick - when no version cane we remove the trailing '/'

File: src/cli/index.ts (hunk 1 above)
Comment (jarrodwatts) [reply src/cli/index.ts:99]: yea its reserved

File: src/cli/index.ts (hunk 2 above)
Comment (jarrodwatts) [reply src/cli/index.ts:118]: not sure what you mean? release/deploy doesn't open automatically for me

Comment (joaquim-verges) [reply src/cli/index.ts:118]: might be a windows thing, but look at what we do in`processor.ts` - we call `open(url)` which attempts to open your default browser

**Comment by jarrodwatts** [comment]
addressed ur feedback @joaquim-verges

**Review by joaquim-verges — APPROVED** [review]
Lgtm

**Pull request merged**